dali/imgcodec/parsers/webp.cc

```diff
@@ -42,27 +42,58 @@ struct RiffHeader {
 };
 static_assert(sizeof(RiffHeader) == 12);
 
-using vp8_header_t = std::array<uint8_t, 4>;
+// Chunk specification:
+// https://developers.google.com/speed/webp/docs/riff_container#riff_file_format
+struct ChunkHeader {
```

Comment:
Could add attribute ``struct [[gnu::packed]]`` ChunkHeader`` where available.

Reply:
To be used along the pragma, yes?

Reply:
@prak-nv how would that play with our clang builds? Can you give us some background on the benefits of using this? It's something we've never used before

Reply:
No it won't affect clang builds on x86_64 might affect powerpc or arm or whatever (depends on platform ABI) though. This attribute ensures that there is no padding for fileds ie. next field will start byte aligned(byte), not with it's own natural alignment;

Please consider `struct A {bool b; int x };` when not packed there will be 3 byte padding since `alignof(x) == sizeof(x) == 4` after field `b` ie. `offsetof(x)` will be 4. When packed `offetof(A::x)` will be 1

Reply:
@staniewzki attribute packed is alternative to `#pragma packed`